### DISCUSSION TITLE
removing static instances of Hibernate types

### DISCUSSION THREAD

**Discussion by gavinking** [hibernate/hibernate-orm#3849] [Design Proposals]
So the ability to write, say, `StringType.INSTANCE` in Hibernate is something so ancient I think it actually predates not just annotations by even _XML_ mappings!

Which means that removing this ability would break some user code.

Nevertheless, I think the amount of problems it's caused in the internal implementation of Hibernate is.... well, uff, I actually would rather not think about that.

And so now that people have had a really *long* time to migrate away from most the older APIs that encourage the use of these static instances, I think we could consider removing them.

Most likely it's only really `UserType`s which are still using them, and migrating `UserType`s is probably not a very painful task for most users.

**Comment by sebersole** [comment]
It will be painful for sure.  But I think it should be done.
